runtime/common/src/lib.rs

```diff
@@ -3,3 +3,33 @@
 pub mod constants;
 pub mod extensions;
 pub mod weights;
+
+/// Macro to set a value (e.g. when using the `parameter_types` macro) to either a production value
+/// or to an environment variable or testing value (in case the `frequency-rococo-local` feature is selected or in instant-sealing mode).
+/// Note that the environment variable is evaluated _at compile time_.
+///
+/// Usage:
+/// ```Rust
+/// parameter_types! {
+/// 	// Note that the env variable version parameter cannot be const.
+/// 	pub LaunchPeriod: BlockNumber = prod_or_local_or_env!(7 * DAYS, 1 * MINUTES, "FRQCY_LAUNCH_PERIOD");
+/// 	pub const VotingPeriod: BlockNumber = prod_or_local_or_env!(7 * DAYS, 1 * MINUTES);
+/// }
+/// ```
+#[macro_export]
+macro_rules! prod_or_local_or_env {
```

Comment:
Slightly modified from https://github.com/paritytech/polkadot/blob/3d6b56342e5e1313685aae6f4e222cbf91ba790f/runtime/common/src/lib.rs#L257

Reply:
why? Can you explain the value add here?

Reply:
Can I do this by building with an environment variable?
// either a production value or to an environment variable or testing value

Reply:
This prevent us from purely relying on env variable and have a default path baked in

Reply:
@enddynayn It is not building with an env, it is running with an env. (See the example in the PR description)